### PR TITLE
ENG-328: Release process part 1

### PR DESCRIPTION
**Pull request by DavidSpek**
## Summary
This PR is automates the release process for our console.

As a first step to make the CI simpler, the `VERSION` file is removed and Elixir is setup to read the release version from git tags.

## Test Plan
Build images locally and test on cluster.

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -61,7 +58,7 @@ fi
 RUN \
   mkdir -p /opt/built && \
   mix distillery.release --name ${APP_NAME} && \
-  cp _build/${MIX_ENV}/rel/${APP_NAME}/releases/${APP_VSN}/${APP_NAME}.tar.gz /opt/built && \
+  cp _build/${MIX_ENV}/rel/${APP_NAME}/releases/*/${APP_NAME}.tar.gz /opt/built && \
```
[the remark this answers is not in the document]
Comment (DavidSpek) [reply Dockerfile:61]: Yep, built and deployed on my cluster.

File: mix.exs [hunk 2]
```diff
@@ -1,12 +1,45 @@
 defmodule Console.MixProject do
   use Mix.Project
 
-  @vsn File.read!("VERSION")
+  defp version do
```
[the remark this answers is not in the document]
Comment (DavidSpek) [reply mix.exs:4]: We can probably remove this entirely. See the explanation of this function [here](https://elixirforum.com/t/psa-using-git-tags-as-library-version-in-hex-packages/29345).

File: mix.exs [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+    case :file.consult('hex_metadata.config') do
+      {:ok, data} ->
+        {"version", version} = List.keyfind(data, "version", 0)
+        version
+      _ ->
+        version =
```
[the remark this answers is not in the document]
Comment (DavidSpek) [reply mix.exs:10]: That makes sense. Will need to dig into Elixir a bit more for that haha

**Comment by DavidSpek** [comment]
Found an issue where the git tag detection isn't available when building the container. Investigating that now.

File: assets/e2e/cypress/utils/graphql-test-utils.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { CyHttpMessages } from 'cypress/types/net-stubbing';
+
+// Utility to match GraphQL mutation based on the operation name
+export const hasOperationName = (req: CyHttpMessages.IncomingHttpRequest, operationName: string) => {
+  const { body } = req
+  return (
+    body.hasOwnProperty('operationName') && body.operationName === operationName
```
Comment (michaeljguarino) [inline assets/e2e/cypress/utils/graphql-test-utils.ts:7]: do we need to use `hasOwnProperty` here?

Comment (DavidSpek) [reply assets/e2e/cypress/utils/graphql-test-utils.ts:7]: Yeah that is what allows us to intercept specific POST requests and use them in `cy.wait(@gqlCreateBuildMutation)` for example.
This code is straight from the Cypress docs for using it with GraphQL.

Comment (michaeljguarino) [reply assets/e2e/cypress/utils/graphql-test-utils.ts:7]: oh I just assumed `body.operationName === operationName` would still be falsy if there's no operationName property

**Review by michaeljguarino — APPROVED** [review]

**Review by maciaszczykm — COMMENTED** [review]
A bit late, but this looks good.

**Pull request merged**